java/memory/src/test/java/org/apache/arrow/memory/TestBaseAllocator.java

```diff
@@ -478,6 +520,79 @@ public void testAllocator_overAllocateParent() throws Exception {
     }
   }
 
+  @Test
+  public void testAllocator_parentLimit_outcomeDetails() throws Exception {
+    try (final RootAllocator rootAllocator =
+        new RootAllocator(MAX_ALLOCATION)) {
+      try (final BufferAllocator childAllocator =
+          rootAllocator.newChildAllocator("child", 0, MAX_ALLOCATION / 2)) {
+        try (final BufferAllocator grandChildAllocator =
+            childAllocator.newChildAllocator("grandchild", MAX_ALLOCATION / 4, MAX_ALLOCATION)) {
+          try {
+            final ArrowBuf arrowBuf = grandChildAllocator.buffer(MAX_ALLOCATION);
+            fail("allocated memory beyond max allowed");
+          } catch (OutOfMemoryException e) {
+
+            // expected
+            AllocationOutcomeDetails outcomeDetails = e.getOutcomeDetails();
+            assertNotNull("missing outcome details", outcomeDetails);
+
+            assertEquals(outcomeDetails.getFailedAllocator(), childAllocator);
+
+            Iterator<Entry> iterator = outcomeDetails.allocEntries.iterator();
+            AllocationOutcomeDetails.Entry first = iterator.next();
+            assertEquals(MAX_ALLOCATION / 4, first.allocatedSize);
+            assertEquals(MAX_ALLOCATION, first.requestedSize);
+            assertEquals(false, first.allocationFailed);
+
+            AllocationOutcomeDetails.Entry second = iterator.next();
+            assertEquals(MAX_ALLOCATION - MAX_ALLOCATION / 4, second.requestedSize);
+            assertEquals(0, second.allocatedSize);
+            assertEquals(true, second.allocationFailed);
+          }
+        }
+      }
+    }
+  }
+
+  @Test
+  public void testAllocator_rootLimit_outcomeDetails() throws Exception {
```

Comment:
failureAtRootLimit?

Reply:
renamed to failureAtRootLimitOutcomeDetails